app/policies/workflow_policy.rb

```diff
@@ -9,4 +9,30 @@ def resolve(action)
   scope :index, :show, :update, :destroy, :update_links, :destroy_links,
         :translate, :versions, :version, :retire_subjects, :create_classifications_export,
         with: Scope
+
+  def linkable_subject_sets
+    # TODO: Any set that is not part of the same project should get duplicated, not linked.
+    # This currently gets done by the controller, and is set up such that this code here
+    # must allow it to be in scope for linking. However, this opens the door for bugs to
+    # accidentally allow linkage of foreign subject sets. It would be much better if the
```

Comment:
what does foreign subject sets mean? is this about mistyping the subject_set_id when linking?

Originally the idea was any public / showable `subject_set` was available for copying.

Reply:
My thinking here was kind of that it might be preferable for this to only return the sets that are already in the current project. Those are the ones actually allowed to be associated with the workflow, as it were.

A user calling the API with a subject set ID that's not in the project, well those aren't allowed to be associated with the workflow. The controller has a special case for those, and it makes a copy of them into the project, and then links up that copy (which is now in the project, and therefore allowed to be linked per the policy).

Note that the controller is already special casing them, by checking if anything was returned by the policy that happens not to be in the project (calling `same_project?` on them), and then copying those. My TODO is to change that so that rather than calling `same_project?`, it'd be more like (not feature-complete, and crudely doing way too many queries, bear with me):

```ruby
workflow.subject_sets = links["subject_sets"].map do |subject_set_id|
  linkable_subject_sets.find(id) ||
    policy_for(SubjectSet).scope_for(:show).find(id).copy_into(project).id
end
```

I left it as a TODO for a future me (I plan to keep making small PRs kind of as friday-afternoon projects or something) because of all the obvious flaws in the above code. Deemed as out of scope for this PR, basically. Happy to move this comment to somewhere else, like a new GH issue. Would probably make more sense.